src/contracts/pods/EigenPodManager.sol

```diff
@@ -207,6 +422,124 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
     }
 
     // INTERNAL FUNCTIONS
+    function _queueWithdrawal(
+        address podOwner,
+        uint256 amountWei,
+        address withdrawer,
+        bool undelegateIfPossible
+    ) 
+        internal
+        returns (bytes32)
+    {
+        require(amountWei > 0, "EigenPodManager._queueWithdrawal: amount must be greater than zero");
+
+        require(amountWei % GWEI_TO_WEI == 0,
+            "EigenPodManager._queueWithdrawal: cannot queue a withdrawal of Beacon Chain ETH for an non-whole amount of gwei");
```

Comment:
do we also ensure that the user's shares only *increase* by a full Gwei amount? if not, seems like we probably should!

Reply:
We use the validator's effective balance to set the shares, which is always denominated in Gwei:

Initially:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/6525091148d1e10a0922c495b74d7c878c6c9d1c/src/contracts/pods/EigenPod.sol#L437

For balance updates, we convert it to Gwei:

https://github.com/Layr-Labs/eigenlayer-contracts/blob/6525091148d1e10a0922c495b74d7c878c6c9d1c/src/contracts/pods/EigenPod.sol#L304

So we're always accountfing for balances in Gwei (hwhich we convert to wei):
initially:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/6525091148d1e10a0922c495b74d7c878c6c9d1c/src/contracts/pods/EigenPod.sol#L445

balance update:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/6525091148d1e10a0922c495b74d7c878c6c9d1c/src/contracts/pods/EigenPod.sol#L320

Reply:
yeah, I mean makes sense. I'm OK w/ merging like this, may add a check that enforces this behavior more strictly but I'll think about it a touch more first.

Reply:
check was added (along with a Prover rule) in a subsequent PR https://github.com/Layr-Labs/eigenlayer-contracts/pull/235